### PR TITLE
[Stats Refresh] Queries Cancelled

### LINKED ISSUE #11742: [Stats Refresh] Period queries being cancelled
A while back, a `cancelQueries` method was added to `StatsPeriodStore` to force cancel queries because they were being re-run when they weren't needed. I.e. overview queries were being re-run when Post Stats was displayed, and vice versa.

However, with the ongoing improvements/changes, I _think_ this is causing some issues.

1. When stats is accessed for the first time, the overview chart is frequently blank.

When this happens, there are a slew of `cancelled` query messages in the logs. Example:

```
2019-05-21 14:41:00:166 WordPress[58418:31284690] Error fetching search terms: Optional("cancelled")
2019-05-21 14:41:00:166 WordPress[58418:31284690] Stats: Finished fetching search terms.
2019-05-21 14:41:00:167 WordPress[58418:31284690] Error fetching referrers: Optional("cancelled")
2019-05-21 14:41:00:167 WordPress[58418:31284690] Stats: Finished fetching referrers.
2019-05-21 14:41:00:167 WordPress[58418:31284690] Error fetching posts: Optional("cancelled")
....

```

2. When accessing a details view (via `View more`), then return to Period overview, several stats frequently show `No data for this period` even though there was data displayed before accessing the details view.

### PR DESCRIPTION
**Pull request by danielebogo**
Fixes #11742 
Refs. #10840

This PR improves the way the Stats views are loaded, avoiding queries being cancelled.
The main change I adopted is using a _UIPageViewController_ (with no datasource and delegate) as container. This avoids to load multiple times the period queries at the beginning. Having 2 containers allocated made the _PeriodStore_ to fetch multiple times the same queries and actions, which generated the cancel of the current fetch.

As I mentioned before the _UIPageViewController_ doesn't have delegate and datasource. This because I don't want any gesture applied and I want to set the current visible view controller only in 2 specific case:
- If the selected period is different (between insights and periods)
- If the _pageViewController_ is empty

@ScoutHarris @stevebaranski I like if you both can have a look to this 😊

## To test:
As @ScoutHarris suggested:
• The last view selected is a Period (so when the app launches and stats is first accessed, it's on the Period view).
• From a new branch, run the app in the same simulator
• When the app launches, it goes directly to stats Period
• You should load the Stats with no problem

**Important:**
This doesn't fix the problem we still have when a Stats detail is pushed. I'm working separately to improve the PeriodStore loading.

Update release notes:

- [ ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

**Comment by ScoutHarris** [comment]
Hey @danielebogo .

Here are some issues I found. 

---
**Period:**
- Stats don't update when a different site is selected.
  - View Period stats for a site.
  - Switch sites.
  - The stats don't update, and still show the stats for the first site.
- The overview chart isn't being updated sometimes, especially for longer periods. (I think the NRV is being removed before the summary query finishes.)
- Pull to refresh doesn't appear to work sometimes. I see these in the logs, and the data doesn't update.
```
2019-05-29 12:05:54:438 WordPress[70071:1073155] Error fetching summary: Optional("cancelled")
2019-05-29 12:05:54:438 WordPress[70071:1073155] Stats: Finished fetching summary.
2019-05-29 12:05:54:438 WordPress[70071:1073155] Error fetching referrers: Optional("cancelled")
2019-05-29 12:05:54:438 WordPress[70071:1073155] Stats: Finished fetching referrers.
2019-05-29 12:05:54:439 WordPress[70071:1073155] Error fetching posts: Optional("cancelled")
2019-05-29 12:05:54:439 WordPress[70071:1073155] Stats: Finished fetching posts.
2019-05-29 12:05:54:439 WordPress[70071:1073155] Error fetching authors: Optional("cancelled")
2019-05-29 12:05:54:439 WordPress[70071:1073155] Stats: Finished fetching authors.
2019-05-29 12:05:54:439 WordPress[70071:1073155] Error fetching videos: Optional("cancelled")
2019-05-29 12:05:54:439 WordPress[70071:1073155] Stats: Finished fetching videos.
2019-05-29 12:05:54:440 WordPress[70071:1073155] Error fetching clicks: Optional("cancelled")
2019-05-29 12:05:54:440 WordPress[70071:1073155] Stats: Finished fetching clicks.
2019-05-29 12:05:54:440 WordPress[70071:1073155] Error fetching countries: Optional("cancelled")
2019-05-29 12:05:54:440 WordPress[70071:1073155] Stats: Finished fetching countries.
2019-05-29 12:05:54:440 WordPress[70071:1073155] Error fetching published: Optional("cancelled")
2019-05-29 12:05:54:440 WordPress[70071:1073155] Stats: Finished fetching published.
2019-05-29 12:05:54:441 WordPress[70071:1073155] Error fetching search terms: Optional("cancelled")
2019-05-29 12:05:54:441 WordPress[70071:1073155] Stats: Finished fetching search terms.
2019-05-29 12:05:54:454 WordPress[70071:1073155] Stats: All fetching operations finished.
```

---
**Insights:**
- Sometimes the LPS chart does not appear. 
  - Seems to be on initial load.  
  - Pull to refresh doesn't fix it.
  - After switching between stats tabs it will eventually appear.

lps_chart

- On a site with very little data, Annual Site Stats shows nothing.

On develop:
annual_develop

On this branch:
annual_branch

File: WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift [hunk 1]
```diff
@@ -117,19 +108,38 @@ private extension SiteStatsDashboardViewController {
...
 
-    func updatePeriodView() {
+    func restoreSelectedPeriodFromUserDefaults() {
+        currentSelectedPeriod = getSelectedPeriodFromUserDefaults()
+    }
 
-        guard currentSelectedPeriod != .insights else {
-            return
+    func updatePeriodView(oldSelectedPeriod: StatsPeriodType) {
+        let selectedPeriosChanged = currentSelectedPeriod != oldSelectedPeriod
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Stats/SiteStatsDashboardViewController.swift]: Nitpick `selectedPeriosChanged` is misspelled.

**Comment by danielebogo** [comment]
Thanks @ScoutHarris ! This is ready for another round!

**Comment by danielebogo** [comment]
Thanks @ScoutHarris !

**Pull request merged**